src/main/java/de/novatec/marioai/tools/MarioAiRunnerBuilder.java

```diff
@@ -0,0 +1,185 @@
+package de.novatec.marioai.tools;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import ch.idsia.ai.agents.Agent;
+import ch.idsia.ai.tasks.ChallengeTask;
+import ch.idsia.ai.tasks.Task;
+/**
+ * @author rgu
+ * @author nmn
+ * 
+ * Builder pattern for MarioAiRunner
+ * 
+ */
+public class MarioAiRunnerBuilder implements Builder{
+	private List<Agent> agents = new ArrayList<Agent>();
+	private LevelConfig levelConfig = LevelConfig.ASTAR_KILLER2;
+	private Task task = new ChallengeTask(); 
+	private int fps = 24; 
+	private int zoomFactor = 3; 
+	private boolean randomize = false; 
+	private boolean viewable = true;
+	private boolean debugView = false;
+	private boolean exitOnFinish = false; 
+	private boolean pushMetrics = false;
+	
+	/**
+	 * @return creates and returns the MarioAiRunner instance using this class' variables
+	 */
+	public MarioAiRunner construct() {
+		return new MarioAiRunner(this);
+	}
+/**
+ * 
+ * @return agents currently supposed to be built later
+ */
+	public List<Agent> getAgents() {
+		return agents;
+	}
+/**
+ * 
+ * @param adds agents to the builder 
+ */
+	public void addAgents(List<Agent> agents) {
+		
+		this.agents.addAll(agents);
+	}
+	/**
+	 * 
+	 * @param adds an agent to the builder
+	 */
+	public void addAgent(Agent agent) {
+		
+		this.agents.add(agent);
+	}
+	
+	/**
+	 * 
+	 * @return LevelConfig supposed to be built
+	 */
+	public LevelConfig getLevelConfig() {
+		return levelConfig;
+	}
+	/**
+	 * 
+	 * @param set levelConfig from the builder
+	 */
+	public void setLevelConfig(LevelConfig levelConfig) {
+		if(levelConfig != null)this.levelConfig = levelConfig;
+	}
+/**
+ * 
+ * @return gets the task from the builder
+ */
+	public Task getTask() {
+		return task;
+	}
+/**
+ * 
+ * @param sets the task in the builder
+ */
+	public void setTask(Task task) {
+		if(task != null)this.task = task;
```

Comment:
Möglicherweise wäre es lesbarer und weniger fehleranfällig wenn selbst bei einzeiligen Statement in einem if- oder Schleifenkörper die geschweiften Klammern genutzt werden.